README.md

```diff
@@ -22,6 +22,36 @@ fn main() {
 }
 ```
 
+# Safety
+
+It is generally not safe to concatenate two adjacent slices.
+This is explained in [#8], [bluss/odds#25], [rust-lang/rust#66111], [rust-lang/rust#62765], [rust-lang/rfcs#2806] and [tokio-rs/bytes#347].
+To sum it up, when rust calculates offsets into a slice, it makes the assumption that elements of that slice stem from the same allocation.
+If two different allocations are adjacent to each other, the `concat*` functions in this crate would readily combine them, breaking internal assumptions of `rustc` and thus may result in UB.
+Therefore, all functions in this crate are marked as `unsafe`.
+The user must ensure that the slices to be concatenated stem from the same allocation (the same `String`, `Vec<_>`, …).
+
+Another edge-case are zero-sized types (ZST).
+Multiple slices over different ZST may point to the same memory-region.
+That is because a ZST doesn't "exist" (i.e. has any size) during runtime.
```

Comment:
How about:

> Instances of ZSTs, including slices, can not be identified by their address even for different types. This is because a ZST does not occupy any space, it does not "alias" any memory.